File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    const set = new Set();
    const length = elements.length;
    const doubled = [...elements, ...elements];
    
    for (let len = 1; len <= length; len++) { 
        for (let i = 0; i < length; i++) {    
            let sum = 0;
            for (let j = i; j < i + len; j++) {
                sum += doubled[j];
            }
            set.add(sum);
        }
    }
    return set.size;
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = 0
    N = len(elements)
    elements = elements + elements
    sums = set()
    
    for i in range(1, N+1):
        for j in range(N):
            sums.add(sum(elements[j:j+i]))
        
    return len(sums)